src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -14755,5 +14755,277 @@ static void Main(List<int> list)
+
+            var comp = CreateCompilation(source).VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "42 43 44 3");
+
+            var program = comp.GetMember<NamedTypeSymbol>("C");
+            var argument = program.GetAttributes().Single().ConstructorArguments.Single();
+            var values = argument.Values;
+            Assert.Equal(3, values.Length);
+            Assert.Equal(42, values[0].Value);
+            Assert.Equal(43, values[1].Value);
+            Assert.Equal(44, values[2].Value);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69133")]
+        public void InAttribute_Empty()
+        {
+            string source = """
+                using System;
+                
+                var attr = (XAttribute)System.Attribute.GetCustomAttribute(typeof(C), typeof(XAttribute));
+                Console.Write(attr._values.Length);
+                
+                [X([])]
+                class C
+                {
+                }
+                
+                public class XAttribute : System.Attribute
+                {
+                    public int[] _values;
+                    public XAttribute(int[] values) { _values = values; }
+                }
+                """;
+
+            var comp = CreateCompilation(source).VerifyEmitDiagnostics( );
+            CompileAndVerify(comp, expectedOutput: "0");
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69133")]
+        public void InAttribute_NotConstant()
+        {
+            string source = """
+                [X([1, 2, C.M()])]
+                class C
+                {
+                    public static int M() => 0;
+                }
+
+                public class XAttribute : System.Attribute
+                {
+                    public XAttribute(int[] values) { }
+                }
+                """;
+
+            var comp = CreateCompilation(source).VerifyEmitDiagnostics(
+                // (1,11): error CS0182: An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type
+                // [X([1, 2, C.M()])]
+                Diagnostic(ErrorCode.ERR_BadAttributeArgument, "C.M()").WithLocation(1, 11)
+                );
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69133")]
+        public void InAttribute_NotConstant_Spread()
+        {
+            string source = """
+                [X([1, 2, .. [3]])]
+                class C
+                {
+                }
+
+                public class XAttribute : System.Attribute
+                {
+                    public XAttribute(int[] values) { }
+                }
+                """;
+
+            CreateCompilation(source).VerifyEmitDiagnostics(
+                // (1,14): error CS9176: There is no target type for the collection expression.
+                // [X([1, 2, .. [3]])]
```

Comment:
definitely an odd error :)

Reply:
I agree. It's an existing problem though and I'm not planning to address in this PR.